### PR TITLE
More integration tests and fixes

### PR DESCRIPTION
**Pull request by Sayan751**
# Pull Request

## 📖 Description

More integration tests involving runtime, and runtime-html. Additionally, there are 2 fixes.

1. Added support for binding object, and array to the `class` CA: https://github.com/aurelia/aurelia/commit/feede3a1009f4f7c46c020554b8e44d06e64bac0
1. Fixed issue with non-removal of old styles in `style-attribute-accessor`: https://github.com/aurelia/aurelia/commit/40db3dcd1097c493c9b10208db8c68ee663cf5a6

### CODE REVIEW

**Comment by EisenbergEffect** [comment]
Please ping me when this is merged. I've got a number of fixups to the style system I want to do once this is in. Waiting for now se we don't get crazy conflicts.

File: packages/runtime-html/src/observation/class-attribute-accessor.ts [hunk 1]
```diff
@@ -97,58 +97,59 @@ export class ClassAttributeAccessor implements IAccessor<unknown> {
...
-    for (let i = 0, length = classes.length; i < length; i++) {
-      const className = classes[i];
-      if (!className.length) {
-        continue;
+  } else if (object instanceof Object) {
+    const classes: string[] = [];
+    for (const property in object) {
+      // Let non typical values also evaluate true so disable bool check
+      // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions, no-extra-boolean-cast
+      if (!!object[property]) {
```
[the remark this answers is not in the document]
Comment (Sayan751) [reply packages/runtime-html/src/observation/class-attribute-accessor.ts]: Done

File: packages/__tests__/integration/app/molecules/index.ts [hunk 2]
```diff
@@ -19,7 +37,16 @@ export const molecules = {
...
-  }
-};
+    },
+    customize(cb?: MolecularConfigCustomizer) {
+      return createMolecularConfiguration(cb || customizeConfiguration);
+    },
+  };
+}
+
+export const molecules = createMolecularConfiguration(() => {/* noop */ });
```
[the remark this answers is not in the document]
Comment (Sayan751) [reply packages/__tests__/integration/app/molecules/index.ts]: Done

File: packages/runtime-html/src/observation/class-attribute-accessor.ts [hunk 3]
```diff
@@ -97,58 +97,59 @@ export class ClassAttributeAccessor implements IAccessor<unknown> {
     }
   }
 
-  private splitClassString(classString: string): string[] {
+  private addClassesAndUpdateIndex(classes: string[]) {
+    const node = this.obj;
+    for (let i = 0, length = classes.length; i < length; i++) {
```
[the remark this answers is not in the document]
Comment (Sayan751) [reply packages/runtime-html/src/observation/class-attribute-accessor.ts]: Done

File: packages/runtime-html/src/observation/class-attribute-accessor.ts [hunk 4]
```diff
@@ -97,58 +97,59 @@ export class ClassAttributeAccessor implements IAccessor<unknown> {
...
+      const className = classes[i];
+      if (!className.length) {
```
[the remark this answers is not in the document]
Comment (Sayan751) [reply packages/runtime-html/src/observation/class-attribute-accessor.ts]: Done

File: packages/__tests__/tasks/copy-html.js [hunk 5]
```diff
@@ -71,7 +71,7 @@ function handleChange(eventName, src) {
   }
 }
 
-const watched = htmlSourceDirs.map((dir) => path.join(cwd, dir, '**/*.html'));
+const watched = sourceDirs.reduce((acc, dir) => [...acc, path.join(cwd, dir, '**/*.html'), path.join(cwd, dir, '**/*.css')], []);
```
[the remark this answers is not in the document]
Comment (Sayan751) [reply packages/__tests__/tasks/copy-html.js]: Done

File: packages/__tests__/integration/integration.spec.ts [hunk 6]
```diff
@@ -810,4 +815,122 @@ describe('app', function () {
...
+    { useCSSModule: true, selectedHeaderColor: 'rgb(0, 0, 255)', selectedDetailsColor: 'rgb(203, 203, 203)' },
+  ].map(({ useCSSModule, selectedHeaderColor, selectedDetailsColor }) =>
+    $it(`uses cards to display topic details which marks the selected topic with a specific color - useCSSModule:${useCSSModule}`,
+      async function ({ host, ctx }) {
+        const container1 = host.querySelector('cards #cards1');
+        const container2 = host.querySelector('cards #cards2');
+        const cards1 = toArray(container1.querySelectorAll('div'));
+        const cards2 = toArray(container2.querySelectorAll('div'));
+
+        assert.equal(getComputedStyle(container1).display, 'flex', 'incorrect container1 display');
```
[the remark this answers is not in the document]
Comment (Sayan751) [reply packages/__tests__/integration/integration.spec.ts]: Done

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://34138-121594173-gh.circle-artifacts.com/0/home/circleci/repo/test/js-framework-benchmark/webdriver-ts-results/table.html

**Pull request merged**